addons/mail/static/src/js/discuss.js

```diff
@@ -195,6 +195,11 @@ var Discuss = AbstractAction.extend(ControlPanelMixin, {
 
     /**
      * @override
+     * @param {Object} [options]
+    //  * @param {boolean} [options.phantomjs=false] if set, rendering of non-empty
+    //  *   thread will use 'block' display instead of 'flex', because phantomjs
+    //  *   sucks with flexbox... TODO: remove this option when we no longer use
+    //  *   phantomJS
```

Comment:
Nop, it isn't. The issue is that phantomJS messes up a test because there is a business logic that is triggered at the moment of the rendering.

Since we need the whole logic of Discuss in the test but adapt it with phantomJS at rendering, here are the 2 options:

1. Put logic for phantomJS in Discuss for the rendering of thread.
2. Split template rendering in many small "rendering" functions in the widget, and patch in the test for phantomJS before it is added in the DOM.

I have opted for option 1, because the diff is much shorter and it is easier to revert when we no longer use phantomJS.